### PR TITLE
feat: LT-974 - Not Yet Verified Good (System Flag)

### PR DESCRIPTION
**Pull request by andytaylorkainos**

### CODE REVIEW

File: flags/tests/tests_list.py [hunk 1]
```diff
@@ -15,16 +13,19 @@ def test_gov_user_can_see_all_flags(self):
...
+        flag4 = self.create_flag("Flag4", "Case", self.team)
 
         response = self.client.get(self.url + "?level=Case&team=" + self.team.name, **self.gov_headers)
 
         response_data = response.json()
         self.assertEqual(response.status_code, status.HTTP_200_OK)
-        self.assertEqual(len(response_data["flags"]), (system_flag_count + 2))
+        returned_flags = [flag["id"] for flag in response_data["flags"]]
+        self.assertIn(str(flag1.id), returned_flags)
+        self.assertNotIn(str(flag2.id), returned_flags)
```
Comment (BenAndrew-Kainos) [inline flags/tests/tests_list.py]: Could you leave a docstring to explain why some are returned and others are not

File: goods/tests/tests_create.py [hunk 2]
```diff
@@ -65,6 +66,8 @@ def test_when_creating_a_good_with_not_controlled_and_not_pv_graded_then_created
         self.assertEquals(response.status_code, status.HTTP_201_CREATED)
         _assert_response_data(self, response.json()["good"], self.request_data)
```
Comment (BenAndrew-Kainos) [inline goods/tests/tests_create.py]: As these 4 lines are repeated multiple times could you please move to a private method

File: flags/tests/tests_list.py (hunk 1 above)
Comment (andytaylorkainos) [reply flags/tests/tests_list.py]: I changed the name of the test to be a bit more specific

File: goods/tests/tests_create.py (hunk 2 above)
Comment (andytaylorkainos) [reply goods/tests/tests_create.py]: Yeah sure!

File: flags/tests/tests_list.py [hunk 3]
```diff
@@ -1,30 +1,31 @@
...
     url = reverse("flags:flags")
 
     def test_gov_user_can_see_all_flags(self):
         response = self.client.get(self.url, **self.gov_headers)
 
         self.assertEqual(response.status_code, status.HTTP_200_OK)
 
-    def test_gov_user_can_see_filtered_flags(self):
-        system_flag_count = Flag.objects.all().count()
+    def test_gov_user_can_see_only_filtered_case_level_and_team_flags(self):
```
Comment (BenAndrew-Kainos) [inline flags/tests/tests_list.py]: Could you still leave a quick docstring to explain why 2 are hidden

File: goods/tests/tests_submitting_goods.py [hunk 4]
```diff
@@ -19,7 +20,10 @@ def test_submitted_good_changes_status(self):
 
         self.client.put(url, **self.exporter_headers)
 
-        self.assertEqual(Good.objects.get().status, "submitted")
+        good = Good.objects.get()
+        self.assertEqual(good.status, "submitted")
+        flags_on_good = [str(id) for id in good.flags.values_list("id", flat=True)]
```
Comment (BenAndrew-Kainos) [inline goods/tests/tests_submitting_goods.py]: Could you update this to use _assert_not_verified_flag_is_set_on_good

File: queries/goods_query/tests.py [hunk 5]
```diff
@@ -105,6 +106,8 @@ def test_respond_to_control_list_classification_query_without_updating_control_c
         self.assertEqual(self.query.good.control_code, previous_query_control_code)
         self.assertEqual(self.query.good.is_good_controlled, str(self.data["is_good_controlled"]))
         self.assertEqual(self.query.good.status, GoodStatus.VERIFIED)
+        flags_on_good = [str(id) for id in self.query.good.flags.values_list("id", flat=True)]
```
Comment (BenAndrew-Kainos) [inline queries/goods_query/tests.py]: Could you update all of these to use _assert_not_verified_flag_is_set_on_good

File: goods/tests/tests_submitting_goods.py (hunk 4 above)
Comment (andytaylorkainos) [reply goods/tests/tests_submitting_goods.py]: Done

File: queries/goods_query/tests.py (hunk 5 above)
Comment (andytaylorkainos) [reply queries/goods_query/tests.py]: Done

File: flags/tests/tests_list.py (hunk 3 above)
Comment (andytaylorkainos) [reply flags/tests/tests_list.py]: Done

File: goods/tests/tests_flags.py [hunk 6]
```diff
@@ -71,7 +74,8 @@ def test_user_can_add_good_level_flags_from_their_own_team(self):
 
         self.client.put(self.good_flag_url, data, **self.gov_headers)
 
-        self.assertEquals(len(data["flags"]), len(self.good.flags.all()))
+        self.assertEquals(len(data["flags"]), 1)
```
Comment (BenAndrew-Kainos) [inline goods/tests/tests_flags.py]: Please also check the return from the PUT

File: goods/tests/tests_submitting_goods.py [hunk 7]
```diff
@@ -19,7 +20,9 @@ def test_submitted_good_changes_status(self):
 
         self.client.put(url, **self.exporter_headers)
 
-        self.assertEqual(Good.objects.get().status, "submitted")
+        good = Good.objects.get()
```
Comment (BenAndrew-Kainos) [inline goods/tests/tests_submitting_goods.py]: Please check returned data from PUT

File: applications/libraries/goods_on_applications.py [hunk 8]
```diff
@@ -0,0 +1,29 @@
...
+    except GoodOnApplication.DoesNotExist:
+        raise Http404
+
+
+def update_good_statuses_and_flags_on_application(application: BaseApplication):
+    if application.application_type == ApplicationType.STANDARD_LICENCE:
+        for good_on_application in GoodOnApplication.objects.filter(application=application):
+            if good_on_application.good.status == GoodStatus.DRAFT:
+                good_on_application.good.status = GoodStatus.SUBMITTED
+                if not good_on_application.good.flags.filter(id=SystemFlags.GOOD_NOT_YET_VERIFIED_ID).exists():
```
[the remark this answers is not in the document]
Comment (andytaylorkainos) [reply applications/libraries/goods_on_applications.py]: Done!

File: applications/libraries/goods_on_applications.py [hunk 9]
```diff
@@ -0,0 +1,29 @@
...
+
+def get_good_on_application(pk):
+    try:
+        return GoodOnApplication.objects.get(pk=pk)
+    except GoodOnApplication.DoesNotExist:
+        raise Http404
+
+
+def update_good_statuses_and_flags_on_application(application: BaseApplication):
+    if application.application_type == ApplicationType.STANDARD_LICENCE:
```
[the remark this answers is not in the document]
Comment (andytaylorkainos) [reply applications/libraries/goods_on_applications.py]: Done

File: goods/tests/tests_submitting_goods.py (hunk 7 above)
Comment (andytaylorkainos) [reply goods/tests/tests_submitting_goods.py]: As discussed, the response doesn't contain anything related to the good. But ive made sure that the response does contain the updated application status

File: common/libraries.py [hunk 10]
```diff
@@ -0,0 +1,36 @@
...
+                "required": strings.Picklists.REQUIRED_REPORT_SUMMARY,
+                "null": strings.Picklists.REQUIRED_REPORT_SUMMARY,
+            },
+        )
+
+
+def update_good_or_goods_type_control_code_details(instance, validated_data):
+    instance.comment = validated_data.get("comment")
+    if validated_data["report_summary"]:
+        instance.report_summary = validated_data.get("report_summary").text
```
Comment (BenAndrew-Kainos) [inline common/libraries.py]: No need to do .get() as well

File: common/libraries.py [hunk 11]
```diff
@@ -0,0 +1,36 @@
+from rest_framework import serializers
```
Comment (BenAndrew-Kainos) [inline common/libraries.py]: Maybe move to goods (goods/libraries)

File: common/libraries.py (hunk 10 above)
Comment (andytaylorkainos) [reply common/libraries.py]: nice find, updated

File: common/libraries.py [hunk 12]
```diff
@@ -0,0 +1,36 @@
...
+from conf.helpers import str_to_bool
+from conf.serializers import ControlListEntryField
+from flags.enums import SystemFlags
+from lite_content.lite_api import strings
+from picklists.models import PicklistItem
+
+
+def initialize_good_or_goods_type_control_code_serializer(self):
+    if str_to_bool(self.get_initial().get("is_good_controlled")):
+        self.fields["control_code"] = ControlListEntryField(required=True, write_only=True)
```
Comment (efahy) [inline common/libraries.py]: `GoodsTypeSerializer` also sets up the `control_code` field this way - can you have it use this common code as well?

Comment (efahy) [reply common/libraries.py]: Also, in `ClcControlGoodTypeSerializer` and `ClcControlGoodSerializer`, the `control_code` field is initially defined as a `CharField`.  Should it not always be a `ControlListEntryField`, but here we just update it to be required?

File: cases/libraries/get_destination.py [hunk 13]
```diff
@@ -64,9 +65,16 @@ def get_ordered_flags(case: Case, team: Team):
     destination_flags = []
 
     if case.type in [CaseTypeEnum.APPLICATION, CaseTypeEnum.HMRC_QUERY, CaseTypeEnum.END_USER_ADVISORY_QUERY]:
-        goods_on_application = GoodOnApplication.objects.filter(application=case).select_related("good")
-        for good_on_application in goods_on_application:
-            goods_flags += good_on_application.good.flags.all()
+        goods_on_application = GoodOnApplication.objects.filter(application=case)
```
[the remark this answers is not in the document]
Comment (andytaylorkainos) [reply cases/libraries/get_destination.py]: As discussed, I will be moving things to the case-model after this story is completed :)

File: common/libraries.py (hunk 12 above)
Comment (andytaylorkainos) [reply common/libraries.py]: I had a look at `GoodsTypeSerializer`
The setup is a little different so I cant use this common code sadly:
```        
       if str_to_bool(self.get_initial().get("is_good_controlled")) is True:
            self.fields["control_code"] = ControlListEntryField(required=True)
        else:
            if hasattr(self, "initial_data"):
                self.initial_data["control_code"] = None
```

Comment (andytaylorkainos) [reply common/libraries.py]: I did however update the `control_code` field in the serializers to use `ControlListEntryField(required=False, ....)` for those serializers as default :)

**Pull request merged**